src/grt/src/Net.h

```diff
@@ -52,11 +54,18 @@ class Net
   const char* getConstName() const;
   odb::dbSigType getSignalType() const;
   void addPin(Pin& pin);
+  void deleteSegment(int seg_id, GRoute& routes);
   std::vector<Pin>& getPins() { return pins_; }
   int getNumPins() const { return pins_.size(); }
   float getSlack() const { return slack_; }
   void setSlack(float slack) { slack_ = slack; }
   void setHasWires(bool in) { has_wires_ = in; }
+  void setSegmentParent(std::vector<uint16_t> segment_parent)
+  {
+    segment_parent_ = std::move(segment_parent);
+  }
+  std::vector<uint16_t> getSegmentParent() const { return segment_parent_; }
+  std::vector<std::vector<uint16_t>> getSegmentGraph();
```

Comment:
Maybe `buildSegmentsGraph`? It makes it clearer that the graph is related to all segments of the net and that this function is creating the graph, not just returning the content of a class member.